staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go

```diff
@@ -379,3 +379,38 @@ func TestPreserveInt(t *testing.T) {
 		t.Errorf("Expected %v, got %v, %v", `9223372036854775807, 1000000`, num1, num2)
 	}
 }
+
+func TestNameCollision(t *testing.T) {
+	stopCh, apiExtensionClient, clientPool, err := testserver.StartDefaultServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer close(stopCh)
+
+	noxuDefinition := testserver.NewNoxuCustomResourceDefinition(apiextensionsv1alpha1.NamespaceScoped)
+	_, err = testserver.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, clientPool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	noxu2Definition := testserver.NewNoxu2CustomResourceDefinition(apiextensionsv1alpha1.NamespaceScoped)
+	_, err = testserver.CreateNewCustomResourceDefinition(noxu2Definition, apiExtensionClient, clientPool)
+	if err == nil {
+		t.Fatalf("unexpected non-error: names chosen for this CustomResourceDefinition conflict with others in the group")
+	}
+
+	err = testserver.DeleteCustomResourceDefinition(noxuDefinition, apiExtensionClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	crd, err := testserver.GetCustomResourceDefinition(noxu2Definition, apiExtensionClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO: Make it pass here
+	if crd.Status.Conditions[0].Status == "True" {
```

Comment:
`Conditions[0]` is not safe. There might be more conditions. We have to go through all conditions to find the right one. @deads2k wrote some helpers (for the internal version only though I fear).